controllers/system/system_controller.go

```diff
@@ -1080,6 +1080,22 @@ func (r *SystemReconciler) ReconcileSystemFinal(client *gophercloud.ServiceClien
 	return nil
 }
 
+// Removes the extra certs from the currentCerts to the specCerts
+func RemoveRuntimeCerts(specCerts, currentCerts *starlingxv1.CertificateList) starlingxv1.CertificateList {
+	var res starlingxv1.CertificateList
+	certificateMap := make(map[string]bool)
+
+	for _, cert := range *specCerts {
+		certificateMap[cert.Secret] = true
```

Comment:
Can you add a comment regarding why to check based on the secret rather than the signature?

Reply:
Explained